cloudflare/workers-sdk
# wrangler: Add source map support for Workers

- Add `upload_source_maps` boolean config option
- Include source maps in Worker upload when enabled

## What this PR solves / how to test

Closes WP-821 and WP-822. Allows users to enable source map support for Workers when using Wrangler. For more context see "SPEC: Workers Source maps" on the internal wiki.

### How to Test

1. Ask @maxwellpeterson to ungate source maps support for your account.
2. Deploy a Worker with `upload_source_maps = true` using the pre-release build for this PR.
3. View uploaded contents using the [Download Worker](https://developers.cloudflare.com/api/operations/worker-script-download-worker) API endpoint.
4. Uploaded contents should include all source maps output by the build step, and each source map should include `sources` and `sourceRoot` values that match the project directory structure.

## Author has addressed the following

- Tests
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
- Public documentation
  - [x] Not necessary because: This feature is not publicly available yet.
